external/upstream/gauxc/CMakeLists.txt

```diff
@@ -0,0 +1,64 @@
+if(${ENABLE_gauxc})
+    if((NOT "${gauxc_DIR}" STREQUAL "") AND "${IntegratorXX_DIR}" STREQUAL "")
+       set(IntegratorXX_DIR ${gauxc_DIR}/../IntegratorXX)
+    endif()
+
+    find_package(ExchCXX CONFIG PATHS ${gauxc_DIR}/../ExchCXX) 
+    find_package(gauxc 0.0.1 CONFIG) 
+
+    if(${gauxc_FOUND})
+        #== check if GPU-enabled GauXC is found, if GPU is specified ==#
+        get_property(_gpu TARGET gauxc::gauxc PROPERTY GAUXC_HAS_CUDA)
+        
+        if(${gauxc_ENABLE_GPU} AND (NOT ${_gpu}))
```

Comment:
Thanks for the heads-up! I suppose this brings up a question on my end, however. If we did only want to support the CUDA version, we would want to use `GAUXC_HAS_CUDA` instead of `GAUXC_HAS_DEVICE` for the definition of `${_gpu}`, right?

Reply:
After looking at this comment, I'm not 100% what I intended here. The breakdown should be as follows:
- Any code-space feature checks should be on `GAUXC_HAS_DEVICE`. You gain nothing by `GAUXC_HAS_CUDA` or `GAUXC_HAS_HIP` beyond intentionally disabling features that are outside your control. Depending on these variables will just make your life more difficult later.
- `GAUXC_ENABLE_CUDA` / `GAUXC_ENABLE_HIP` should really be the only things used that are backend specific - GauXC will populate `GAUXC_HAS_DEVICE` accordingly (the user-facing code is agnostic to the backend). 


I think that might be what I meant here: for now, you can do what you're already doing (i.e. delegating (`GAUXC_ENABLE_CUDA` to `gauxc_ENABLE_GPU`). All other checks should be on `GAUXC_HAS DEVICE`

Reply:
Thanks! All right, I've switched over all `GAUXC_HAS_CUDA` instances to `GAUXC_HAS_DEVICE`.